libs/wingc/src/type_check.rs

```diff
@@ -2211,7 +2221,9 @@ impl<'a> TypeChecker<'a> {
 				let extend_interfaces = extends
 					.iter()
 					.filter_map(|i| {
-						let t = resolve_user_defined_type(i, env, stmt.idx).unwrap_or_else(|e| self.type_error(e));
+						let t = self
+							.resolve_user_defined_type(i, env, stmt.idx)
+							.unwrap_or_else(|e| self.type_error(e));
```

Comment:
I wonder if there's any way to make sure that the method version of `self.resolve_user_defined_type` always gets used instead of the free-floating function

Reply:
they actually both serve different purposes now.
The method version not also imports the type from JSII as needed, whereas the function version will fail if any types are not already imported.

I'll add docstrings to clarify